source/FreeRTOS_DNS_Cache.c

```diff
@@ -284,8 +348,14 @@
                 xDNSCache[ uxIndex ].ucCurrentIPAddress++;
             #endif /* if ( ipconfigDNS_CACHE_ADDRESSES_PER_ENTRY > 1 ) */
 
-            *pulIP = xDNSCache[ uxIndex ].ulIPAddresses[ ulIPAddressIndex ];
+            ( void ) memcpy( pxIP, &( xDNSCache[ index ].xAddresses[ ulIPAddressIndex ] ), sizeof( *pxIP ) );
             isRead = pdTRUE;
+
+            if( ppxAddressInfo != NULL )
+            {
+                /* Copy all entries from position 'index' to a linked struct addrinfo. */
+                prvReadDNSCache( index, ppxAddressInfo );
```

Comment:
`index` is not defined
~~~diff
-    /* Copy all entries from position 'index' to a linked struct addrinfo. */
-    prvReadDNSCache( index, ppxAddressInfo );
+    /* Copy all entries from position 'uxIndex' to a linked struct addrinfo. */
+    prvReadDNSCache( uxIndex, ppxAddressInfo );
~~~